sympy/matrices/expressions/matpow.py

```diff
@@ -56,14 +56,25 @@ def doit(self, **kwargs):
             args = self.args
         base = args[0]
         exp = args[1]
-        if isinstance(base, MatrixBase) and exp.is_number:
+        if exp.is_zero and base.is_square:
+            try:
+                b = base.__class__(Identity(base.shape[0]))
+                if b == Identity(base.shape[0]).as_explicit():
+                    return b
+                else:
+                    return Identity(base.shape[0])
+            except TypeError:
+                return Identity(base.shape[0])
+        elif isinstance(base, ZeroMatrix) and exp < 0:
+            raise ValueError("Matrix det == 0; not invertible.")
+        elif isinstance(base, (Identity, ZeroMatrix)) and exp.is_number:
```

Comment:
I think for identities matrices, it should always return the base. For zero matrices, if should return the base if the exponent is positive, the identity if it is zero, raise ValueError if it is negative, and remain unevaluated if the sign cannot be determined.